src/controls/PolyControl.ts

```diff
@@ -0,0 +1,153 @@
+import { Point } from '../point.class';
+import { Control } from './control.class';
+import { TMat2D } from '../typedefs';
+import { iMatrix } from '../constants';
+import { Polyline } from '../shapes/polyline.class';
+import { multiplyTransformMatrices } from '../util/misc/matrix';
+import { TPointerEvent, Transform } from '../EventTypeDefs';
+import { getLocalPoint } from './util';
+import { degreesToRadians } from '../util/misc/radiansDegreesConversion';
+import { TransformActionHandler } from '../EventTypeDefs';
+
+export class PolyControl extends Control {
+  pointIndex: number;
+  actionName = 'modifyPolygon';
+
+  constructor(options: Partial<Control>, pointIndex: number) {
+    super(options);
+    this.pointIndex = pointIndex;
+  }
+
+  static applySkew(point: Point, shear: Point) {
+    const skewedPoint = new Point();
+    skewedPoint.y = point.y + point.x * shear.y;
+    skewedPoint.x = point.x + skewedPoint.y * shear.x;
+    return skewedPoint;
+  }
+
+  static removeSkew(point: Point, shear: Point) {
+    const unskewedPoint = new Point();
+    unskewedPoint.x = point.x - point.y * shear.x;
+    unskewedPoint.y = point.y - unskewedPoint.x * shear.y;
+    return unskewedPoint;
+  }
+
+  static getSize(poly: Polyline) {
+    return new Point(poly.width, poly.height);
+  }
+
+  /**
+   * This function locates the controls.
+   * It'll be used both for drawing and for interaction.
+   */
+  positionHandler(dim: Point, finalMatrix: TMat2D, polyObject: Polyline) {
+    const x = polyObject.points[this.pointIndex].x - polyObject.pathOffset.x,
+      y = polyObject.points[this.pointIndex].y - polyObject.pathOffset.y;
+    return new Point(x, y).transform(
+      multiplyTransformMatrices(
+        polyObject.canvas?.viewportTransform ?? iMatrix,
```

Comment:
canvas is always defined in when running a control action
so I guess `polyObject.canvas!.viewportTransform` is enough.
but perhaps a better fix would be to assert the type in the control class

Reply:
take a look d9195a406

Reply:
https://codesandbox.io/s/custom-control-poly-fix-flip-forked-3k408w?file=/src/index.js